File: tests/resources/metrics_sample.php
<?php
// A sample used for metrics.
require_once "vendor/autoload.php";

/* A block comment
   spanning lines. */
function classify(array $values): int {
	$score = 0;
	foreach ($values as $value) {
		if ($value > 10) {
			$score += $value;
		} elseif ($value < 0) {
			continue;
		} else {
			$score--;
		}
	}
	for ($i = 0; $i < 3; $i++) {
		$score++;
	}
	while ($score > 100) {
		$score -= 50;
	}
	switch ($score % 3) {
		case "zero":
			$score++;
			break;
		case "one":
			$score += 2;
			break;
		default:
			$score += 3;
	}
	return $score;
}

echo classify([1, 20, -5]);

File: tests/resources/metrics_sample.go
// Package main is a sample used for metrics.
package main

import "fmt"

/* A block comment
   spanning lines. */
func classify(values []int) int {
	score := 0
	for i, v := range values {
		if v > 10 {
			score += i
		} else if v < 0 {
			continue
		} else {
			score--
		}
	}
	for j := 0; j < 3; j++ {
		switch j {
		case 0:
			score++
		case 1:
			score += 2
		default:
			score += 3
		}
	}
	defer fmt.Println("done")
	if score > 100 {
		goto end
	}
	return score
end:
	return -1
}

func main() {
	fmt.Println(classify([]int{1, 20, -5}))
}
